File: Layouts/app/src/main/java/com/uabc/amc/layouts/MainActivity.java
package com.uabc.amc.layouts;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void changeContentViewLayout(String type) {
        Intent intent = new Intent(this, ExampleLayout.class);
        intent.putExtra(ExampleLayout.ACTIVITY_ACTIVE, type);
        startActivity(intent);
    }

    public void changeLayoutToLinear(View view) {
        changeContentViewLayout("linear");
    }

    public void changeLayoutToGrid(View view) {
        changeContentViewLayout("grid");
    }

    public void changeLayoutToFrame(View view) {
        changeContentViewLayout("frame");
    }

    public void changeLayoutToExercise1(View view) {
        changeContentViewLayout("exercise_1");
    }

    public void changeLayoutToExercise2(View view) {
        changeContentViewLayout("exercise_2");
    }
}

File: Joke/app/src/main/java/com/uabc/amc/joke/MainActivity.kt
package com.uabc.amc.joke

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onClick(view: View) {
        val intent = Intent(this, DelayedMessageService::class.java)
        intent.putExtra(EXTRA_MESSAGE, resources.getString(R.string.message_notify))
        startService(intent)
    }
}